kairos-prover/kairos-circuit-logic/src/account_trie.rs

```diff
@@ -199,6 +199,16 @@ impl<S: Store<Account>> AccountTrie<S> {
 
         Ok(())
     }
+
+    /// Returns the nonce for an accounts public key if it's known, returns an error if unknown.
+    pub fn get_nonce_for(&self, account: &PublicKey) -> Result<u64, TxnErr> {
+        let [account_hash] = hash_buffers([account]);
+
+        let account = self.txn.get(&account_hash)?;
```

Comment:
Using get adds dependencies to the transaction's snapshot.
Use `get_exclude_from_txn`
```suggestion
        let account = self.txn.get_exclude_from_txn(&account_hash)?;
```

Reply:
done